source-kafka/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Build Stage
+################################################################################
+FROM rust:1.54-slim-buster as builder
+
+RUN rustup component add clippy
+
+RUN apt-get update \
+ && apt-get install -y ca-certificates pkg-config cmake g++ libssl-dev libsasl2-dev \
+ && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /connector
+
+COPY source-kafka/Cargo.* /connector/
+
+# Avoid having to install/build all dependencies by copying the Cargo files and
+# making a dummy src/main.rs and empty lib.rs files.
+RUN mkdir src \
```

Comment:
Is the goal of this just to get cargo to download the dependencies so that they're cached within a docker image layer? If so, then perhaps [cargo fetch](https://doc.rust-lang.org/cargo/commands/cargo-fetch.html) might be better.

Reply:
It also caches the compiled output. It's more to prevent re-compiling the dependencies again if you didn't change them than it is to avoid redownloading them. In comparison, downloading them is pretty quick.

I didn't know/remember `cargo fetch`, but it looks like it _only_ downloads the dependencies. You still have to trigger compilation.